### ISSUE TITLE
Please add the option to upload files chunks in parallel to improve upload speed for OneDrive

### ISSUE THREAD

**Issue by Emasoft** [microsoftgraph/msgraph-sdk-dotnet#1647]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated with the poor upload speed of OneDrive when using the ```LargeFileUploadTask``` as explained in the [MS Documentation](https://learn.microsoft.com/en-us/graph/sdks/large-file-upload?context=graph%2Fapi%2F1.0&view=graph-rest-1.0&tabs=csharp). Even on my fiber adsl, capable of 250Mbps upload speed (≈30Mbytes/s) the file is uploaded at 2-3 Mbyte/s. 
This is true even on platforms other than Windows, like Mac, iOS or Android.

**Describe the solution you'd like**
The solution I propose is to allow multi chunks uploads in parallel with a new Task called ```LargeFileParallelUploadTask```. This should be able to open multiple simultaneous connections to the server and upload many chunks in parallel, filling the available user upload bandwidth.

**Describe alternatives you've considered**
An alternative that my clients mention is to use the OneDrive web client browser, that is for some reason much faster (I tested it and it uploads at a speed of 27.8 Mbyte/s , saturating my upload bandwidth). Some clients are reclutant to use my apps because they are slower compared to the OneDrive web client browser. The only thing that keep them using my apps is the ability to schedule automatic backups, an ability missing from the browser.

**Comment by robinmeure** [comment]
Any update on this?

**Comment by KnapSac** [comment]
@andrueastman would you be interested in a PR adding an alternative upload task which supports chunked uploads? That could serve as a good stepping stone towards a parallel version of that task.

**Comment by andrueastman** [comment]
Holding this for now as the API does not support parallel chunk uploading.
https://learn.microsoft.com/en-us/graph/api/driveitem-createuploadsession?view=graph-rest-1.0&preserve-view=true#upload-bytes-to-the-upload-session
![image](https://github.com/microsoftgraph/msgraph-sdk-dotnet/assets/6464005/05403365-b188-4a3b-8023-7bc8dcbb60fe)
